src/router/index.ts

```diff
@@ -40,8 +52,26 @@ const routes: Array<RouteRecordRaw> = [
 ];
 
 const scrollBehavior: RouterScrollBehavior = (to) => {
-  if (to.hash) return { el: to.hash };
-  else return { left: 0, top: 0 };
+  if (to.hash) {
+    // get target element of hash
+    let target = document.querySelector(window.location.hash);
+    if (!target) return;
+
+    // move target to parent section element if this is first child
+    if (
+      target === target.parentElement?.firstElementChild &&
+      target.parentElement?.tagName
+    )
+      target = target.parentElement;
+
+    // get offset to account for header
+    const offset = document.querySelector("header")?.clientHeight || 0;
+
+    return {
+      el: target,
+      top: offset,
+    };
+  }
```

Comment:
Just a super minor point, but I've noticed that the scroll position isn't reset when you navigate between pages (say, from midway down on the Team page to the About page). There's a bit about how to do that here for Vue router v4: https://next.router.vuejs.org/guide/advanced/scroll-behavior.html.

Specifically, I'd add the `from` and `savedPosition` arguments to this function, then put `else if (savedPosition) { return savedPosition; } else { return { top: 0 }; }` near the end.

Reply:
Yes, I'm in the process of adding that in right now. Also I'm cleaning up this function in general.